pkg/packageinstall/packageinstall.go

```diff
@@ -97,6 +97,10 @@ func (pi *PackageInstallCR) reconcile(modelStatus *reconciler.Status) (reconcile
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	if pkg.Spec.Revoked.Reason != "" {
```

Comment:
i think your else case it way down at the bottom of this function. i think it would be better to keep it here: 

else {
   modelStatus.UnsetPackageRevoked(pi.model.ObjectMeta)
}